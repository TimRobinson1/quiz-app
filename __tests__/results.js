import { Text } from 'react-native';
import React from 'react';
import Results from '../src/components/results.js';
import { shallow } from 'enzyme';
const testScore = function(n) {
  return { state: { params: {score: n} } };
}

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// this.props.navigation.state.params.score

it("renders correctly", () => {
  const tree = renderer.create(<Results navigation={testScore(0)} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("is a test", () => {
  const page = shallow(<Results navigation={testScore(0)} />);
  expect(page.find('.button').node.props.children).toContain('Try Again');
});

it("displays that you're good to go with a very low score", () => {
  const page = shallow(<Results navigation={testScore(0)} />);
  expect(page.find('.result').node.props.children).toContain("You're good to go!");
});

it("displays that you're okay with a low score", () => {
  const page = shallow(<Results navigation={testScore(20)} />);
  expect(page.find('.result').node.props.children).toContain("You're mostly okay.");
});

it("displays that you're tipsy with an average score", () => {
  const page = shallow(<Results navigation={testScore(30)} />);
  expect(page.find('.result').node.props.children).toContain("You're a little tipsy, matey.");
});

it("displays that you're drunk with a high score", () => {
  const page = shallow(<Results navigation={testScore(35)} />);
  expect(page.find('.result').node.props.children).toContain("You're completely smashed.");
});
