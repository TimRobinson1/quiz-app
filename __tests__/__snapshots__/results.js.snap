// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  source={1}
  style={
    Object {
      "flex": 1,
      "height": 50,
      "width": null,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(102,0,102,0.5)",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "justifyContent": "flex-end",
          "padding": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        className="result"
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 40,
            "fontWeight": "bold",
            "marginBottom": 40,
            "textAlign": "center",
          }
        }
      >
        You're good to go!
      </Text>
    </View>
    <View />
    <View
      style={
        Object {
          "flex": 4,
          "padding": 40,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits="button"
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007aff",
                "fontSize": 17,
                "fontWeight": "500",
                "textAlign": "center",
              },
              null,
              Object {
                "backgroundColor": "purple",
                "borderColor": "#FFFFFF",
                "borderRadius": 15,
                "borderWidth": 1,
                "color": "#FFFFFF",
                "fontSize": 30,
                "height": 50,
                "margin": 10,
                "overflow": "hidden",
                "paddingTop": 5,
              },
              null,
            ]
          }
        >
          Try Again
        </Text>
      </View>
    </View>
  </View>
</Image>
`;
